lib/auth/saml.go

```diff
@@ -128,6 +124,44 @@ func (a *Server) CreateSAMLAuthRequest(req services.SAMLAuthRequest) (*services.
 	return &req, nil
 }
 
+func (a *Server) getConnectorAndProvider(ctx context.Context, req services.SAMLAuthRequest) (types.SAMLConnector, *saml2.SAMLServiceProvider, error) {
+	var connector types.SAMLConnector
+	var provider *saml2.SAMLServiceProvider
+	var err error
+
+	if req.SSOTestFlow {
```

Comment:
nit: just a preference but for me the second form is more readable
```go
func name1() (int, int, err) {
	int x
	int z 
	int err
	if ... {
		x,err  = ...
		y,err = ...
	} else {
		x,err  = ...
		y,err = ...
	}
	return x, y , z
}
```
vs
```go
func name2() int, int, err {
	if ... {
		x,err  := ...
		y,err := ...
		return x, y, err
	} 
	x,err  := ...
	y,err := ...
	return x, y , z
}
```

Reply:
I'm not entirely happy with this function either, I'll give it a go.

Reply:
Done.